### PR TITLE
post creation

### PR DESCRIPTION
**Pull request by JDC-horizons**

### CODE REVIEW

File: src/controllers/post.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
 import { sendDataResponse } from '../utils/responses.js'
+import { PrismaClient } from '@prisma/client'
+import jwt from 'jsonwebtoken'
+const prisma = new PrismaClient()
+// ignore
```
Comment (ezeikel) [inline src/controllers/post.js:5]: Do we need this comment?

**Review by ezeikel — APPROVED** [review]

File: src/controllers/post.js (hunk 1 above)
Comment (JDC-horizons) [reply src/controllers/post.js:5]: No, will delete

**Review by th0jensen — APPROVED** [review]

File: src/controllers/post.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import { sendDataResponse } from '../utils/responses.js'
+import { PrismaClient } from '@prisma/client'
+import jwt from 'jsonwebtoken'
+const prisma = new PrismaClient()
```
Comment (dogezen) [inline src/controllers/post.js:4]: don't create a new prisma client, import the globally available prisma client in `/utils/dbClient.js`

File: src/controllers/post.js [hunk 3]
```diff
@@ -7,7 +10,33 @@ export const create = async (req, res) => {
     return sendDataResponse(res, 400, { content: 'Must provide content' })
   }
 
-  return sendDataResponse(res, 201, { post: { id: 1, content } })
+  const auth = req.headers.authorization
```
Comment (dogezen) [inline src/controllers/post.js:13]: don't repeat the token validation code here, there's a middleware function called `validateAuthentication` that should be called in the router, before this controller function. In fact, the validation is already being called in the post router:

```js
router.post('/', validateAuthentication, create)
```

File: src/controllers/post.js [hunk 4]
```diff
@@ -7,7 +10,33 @@ export const create = async (req, res) => {
...
+
+  const token = auth.split(' ')[1]
+
+  if (!token) {
+    return res.status(401).json({ error: 'Not authorised' })
+  }
+
+  try {
+    const decodedToken = jwt.verify(token, process.env.JWT_SECRET)
+    const userId = decodedToken.id
```
Comment (dogezen) [inline src/controllers/post.js:27]: you can get the userId from `req.user.id`

**Review by dogezen — CHANGES_REQUESTED** [review]
1) change where we are loading the prisma client
2) remove the auth code from the controller

**Review by ezeikel — APPROVED** [review]

**Review by dogezen — APPROVED** [review]

**Review by th0jensen — APPROVED** [review]

**Review by ezeikel — APPROVED** [review]

**Pull request merged**